apiserver/facades/client/client/client.go

```diff
@@ -65,12 +65,12 @@ type Client struct {
 
 func (c *Client) checkCanRead() error {
 	isAdmin, err := c.api.auth.HasPermission(permission.SuperuserAccess, c.api.stateAccessor.ControllerTag())
-	if err != nil {
+	if _, is := errors.AsType[*apiservererrors.AccessRequiredError](err); err != nil && !is {
```

Comment:
I'm not sure this is correct - error can be either ErrPerm or AccessRequiredError depending on whether we have a jwt or not. And this should be handled internally in the HasPermission method, not exposed at all the call sites

Reply:
We need to have a discussion here around correct logic. At the moment it is correct but what it is trying to fix was not entirely correct.

Specifically before this change we would only return this structure when the auth method was a JWT but we still would completely throw away the contents of the error for ErrPerm.

Or more importantly we would not be evaluating all of facade permissions checks that ask the question canReadModel that checks if one is super admin or has model read.

Reply:
For future readers of this PR.

With the introduction of this new work we have had to make some new contracts around Juju's RPC api.

Specifically when a client would login and perform an action on the api that they were not authorised to do so Juju would return ErrPerm.

Now with JWT's we need to return a different type of error that describes the missing permissions the user has. With these two patterns what we don't want to do is sprinkle if then else logic in our facades based on the authentication method the current entity is using.

The only easy way to achieve this at the moment is to pass off the responsibility of the error decorating to the authentication provider directly.